Add RH feedback module to all relevant guides

mdolezelova opened the pull request:
* [ ] I am familiar with the [contributing](https://github.com/theforeman/foreman-documentation/blob/master/CONTRIBUTING.md) guidelines.

Please cherry-pick my commits into:

* [x] Foreman 3.3/Katello 4.5
* [x] Foreman 3.2/Katello 4.4
* [x] Foreman 3.1/Katello 4.3
* For Foreman 3.0 or older, please create a separate PR.
* We do not accept PRs for Foreman 2.3 or older.

mdolezelova commented:
Hello, this PR is to add the RH feedback module to all relevant guides. The purposed is to guide RH customer on how to provide feedback on docs. We need to add this as we've enabled the DDF (Direct Documentation Feedback) tool for RH Satellite 6.11 and we assume from now it will be enabled for all upcoming releases.
 @Lennonka , please could you help with the merge?
@vferschm, please could you review the PR?
Thank you!

evgeni commented on guides/common/modules/proc_providing-feedback-on-red-hat-docs.adoc:
```diff
@@ -0,0 +1,24 @@
+[l10n-exclude-end]
+* For submitting feedback via Bugzilla, create a new ticket:
++
+. Go to the link:https://bugzilla.redhat.com/enter_bug.cgi?product=Red%20Hat%20Enterprise%20Linux%209[Bugzilla] website.
```
This doesn't look like the right link for Satellite ;-)

evgeni commented on guides/common/modules/proc_providing-feedback-on-red-hat-docs.adoc:
```diff
@@ -0,0 +1,24 @@
+:_content-type: PROCEDURE
+
+[preface]
+[id="proc_providing-feedback-on-red-hat-documentation_{context}"]
```
`missing attribute: context`

evgeni replied:
The Configuring_Load_Balancer guide doesn't define a `context` attribute in its `master.adoc` and thus your inclusion fails as it relies on the `context` attribute being present.

The same is true for the RN doc, but I think you didn't edit that one?

```console
% find . -name master.adoc | xargs grep --files-without-match ':context:'
./guides/doc-Release_Notes/master.adoc
./guides/doc-Configuring_Load_Balancer/master.adoc
```

mdolezelova replied:
Hi @evgeni , thank you so much for the guidance!  Got it. So I've added that context to master.adoc for the Load balancing guide in my last commit. You're right, I haven't touched the master.adoc for the RNs (Satellite has RNs written in the downstream repo).
Thank you!

evgeni commented:
Is there any reason not to pick that to 3.2? While there is no Satellite version based on 3.2 (and thus it won't be used in reality) it would make cherry picking *other* changes to `master.adoc` of the various guides easier.

maximiliankolb commented:
> Is there any reason not to pick that to 3.2? While there is no Satellite version based on 3.2 (and thus it won't be used in reality) it would make cherry picking _other_ changes to `master.adoc` of the various guides easier.

Yes! I want to add this to our PR template for ages now. Will do so eventually ...

maximiliankolb commented on guides/common/modules/proc_providing-feedback-on-red-hat-documentation.adoc:
```diff
@@ -0,0 +1,24 @@
+:_content-type: PROCEDURE
+
+[preface]
```
```suggestion
```

mdolezelova replied:
@maximiliankolb , removed the first line. The [preface] must stay there because of the structure of downstream docs. Thanks for understanding!

maximiliankolb reviewed (changes requested):
Please also rebase to master and reword your commit message. See CONTRIBUTING.md for more information.

Lennonka commented on guides/common/modules/proc_providing-feedback-on-red-hat-documentation.adoc:
```diff
@@ -0,0 +1,24 @@
+. Click the *Add Feedback* pop-up that appears below the highlighted text.
+. Follow the displayed instructions.
+
+[l10n-exclude-end]
```
This looks to me it might have something to do with localization (l10n) that we use internally? But I'm wildly guessing...

vferschm replied:
@maximiliankolb @Lennonka This must be part of the module because we translate the Satellite documentation to Japanese and we want to exclude this module from the translation (Example here: https://access.redhat.com/documentation/ja-jp/red_hat_satellite/6.11/html/installing_capsule_server/installing-capsule-server).

mdolezelova commented on guides/common/modules/proc_providing-feedback-on-red-hat-documentation.adoc:
```diff
@@ -0,0 +1,24 @@
+:_content-type: PROCEDURE
+
+[preface]
+[id="proc_providing-feedback-on-red-hat-documentation_{context}"]
```
Hi @maximiliankolb , thank you very much for your suggestions for improvement. I fully understand them. However, in this case, I would like to ask if it would be acceptable to skip them all, please? And merge the module as it is? The purpose of the new module is to inform the users of *downstream* documentation how to provide the feedback on documentation. There are two ways - either the DDF (Direct Documentation Feedback) - they just highlight a piece of text, the "feedback" button appears, they type their feedback, and by submitting it, a Bugzilla ticket is opened automatically. The second way is to open a Bugzilla ticket manually. The module I'm adding is to explain these two ways. The module exactly as it is, is used in all RHEL guides and maybe also in some other RH products. We would like to use exactly the same for RH Satellite. It was not needed before, because the DDF  functionality was not enabled for previous versions of RH Satellite docs. It has been enabled with Satellite 6.11 (it was a long-term plan), so we need to add that module into all guides that are published for RH Satellite. Please, understand the situation that if some Satellite guides are based on upstream repo and some on downstream, it might look quite inconsistent if we use different modules. The explanatory module on how to provide feedback must look the same in all guides. The module is only applicable to Satellite, so it will never appear in upstream documentation. So please could it be accepted as it is? As to the cherry-picking, I thought that if it's needed only for versions relevant to Satellite release, then we don't need to request cherry-picking to 3.2. But if you think it's a safer way to cherry-pick it everywhere, than I don't have any concerns about it and let's cherry-pick it to 3.2 as well. Apologies for any inconvenience. I really appreciate  all your suggestions for improvement, which I would incorporate in a normal case, but please understand that this is a bit exceptional situation. Also, we need to get the instructions on how to use the DDF tool to customers ASAP. Thank you!

maximiliankolb replied:
Hi @mdolezelova 

I did not request a total rework/rewrite, but trivial changes to align your proposed change to our community standards. For more information, see CONTRIBUTING.md or seek help internally.

Regarding cherry-picking: It most likely makes things easier for everyone if we cherry-pick to all versions between 'master' and '3.1' as well.

maximiliankolb commented on guides/doc-Managing_Content/master.adoc:
```diff
@@ -7,6 +7,10 @@ include::common/header.adoc[]
+ifdef::satellite[]
+include::common/modules/proc_providing-feedback-on-red-hat-documentation.adoc[leveloffset=+1]
+endif::[]
+
```
If you add those four lines above line 7, then it's easier to read and we avoid nested "if" directives. I am not saying "remove it".

mdolezelova replied:
@maximiliankolb , moved to line 7 as you've reuested. Apologies, it was copied to a wrong place originally.

mdolezelova commented on guides/common/modules/proc_providing-feedback-on-red-hat-documentation.adoc:
```diff
@@ -0,0 +1,24 @@
+[l10n-exclude-start]
+* For simple comments on specific passages:
++
+. Make sure you are viewing the documentation in the _Multi-page HTML_ format. In addition, ensure you see the *Feedback* button in the upper right corner of the document.
```
@maximiliankolb , applied. Thank you for the suggestion.

mdolezelova commented on guides/common/modules/proc_providing-feedback-on-red-hat-documentation.adoc:
```diff
@@ -0,0 +1,24 @@
+[id="proc_providing-feedback-on-red-hat-documentation_{context}"]
+= Providing feedback on Red Hat documentation
+
+[role="_abstract"]
```
@maximiliankolb , applied. Thank you for the suggestion!

mdolezelova commented:
Hi @maximiliankolb , I've incorporated all changes that I could. Apologies, but the signs for the localization team [l10n-exclude-start] and [l10n-exclude-end] must stay there. I've checked it with the localization team that does the translations. It's important for their automation tools. Also [preface] must stay there. Other than that, I've incorporated everything. I've rebased to master. I've requested cherry-picking to 3.2. I haven't rewritten the commit message - apologies, I don't understand which of  the commit messages you mean and I'm not sure how to rewrite particular commit message that already exists. If it's crucial, please advise.  I see all checks have passed. Please, it would be good if this can merged soon. RH customers and partners are waiting for the instructions on how to use the DDF tool quite desperately. Thank you!

maximiliankolb commented:
@mdolezelova Something went wrong during the rebase.

Please rebase your branch to master and reword the commit message. See CONTRIBUTING for more information or seek help internally.

Then I or someone else can re-review your actual change.

mdolezelova commented:
Hi @maximiliankolb , after  "git rebase master" on my topic branch, it says "Successfully rebased and updated refs/heads/Add_feedback_module." After checking with "git status", it says nothing to commit.  I don't know what might have gone wrong :-( I may need to close this PR and open a new one where I incorporate your suggestions directly.

maximiliankolb commented:
@mdolezelova I am very confident that this can be recovered; maybe with a little help from someone more experienced with git.

mdolezelova commented:
Hi @maximiliankolb , okay, I understand that requirements for repo contributions must always be met 100%. I thought the process can be speed up a bit in exceptional and urgent cases like this. But sticking to contribution guidelines is completely fair.  I understand it. 
I've done rebase in the past and tried to apply the same way now. Not sure why it doesn't work now.  So let's wait for @Lennonka to be online and help with the rebase. Thank you in advance, Lena!

mdolezelova commented:
Hi @maximiliankolb , Lena helped me to do the rebase correctly and to rewrite the commit message (thanks again, Lena :-)). Please, could you review again to confirm whether the PR is okay to be merged? Thank you!

maximiliankolb reviewed (approved):
LGTM!

mdolezelova commented:
Thank you, @maximiliankolb ! :-)
